File: cpp/Problem344.cpp
/*
 * Problem: 344. Reverse String
 * URL: https://leetcode.com/problems/reverse-string/
 * Difficulty: Easy
 * Tags: Array, String, Reverse
 */

#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
	void reverseString(vector<char>& s) {
		for (int i = 0, j = s.size() - 1; i < j; i++, j--) {
			swap(s[i], s[j]);
		}
	}
};

File: python/Problem344.py
# Problem: 344. Reverse String
# URL: https://leetcode.com/problems/reverse-string/
# Difficulty: Easy
# Tags: Array, String, Reverse

from typing import List

class Solution:
  def reverseString(self, s: List[str]) -> None:
    i: int = 0
    j: int = len(s) - 1
    while i < j:
      s[i], s[j] = s[j], s[i]
      i += 1
      j -= 1